### PR TITLE
Updated e2e setup

### PR DESCRIPTION
**Pull request by rossf7**
Towards giantswarm/giantswarm#4949

The e2e setup code here was lagging behind. Getting it straight here for the existing chartconfig CRD tests so I can add the new chart CRD tests.

### CODE REVIEW

File: integration/chartconfig/chartconfig.go [hunk 1]
```diff
@@ -14,21 +14,20 @@ import (
...
 	"github.com/spf13/afero"
 	"k8s.io/helm/pkg/helm"
 
 	"github.com/giantswarm/chart-operator/integration/env"
+	"github.com/giantswarm/chart-operator/integration/setup"
 	"github.com/giantswarm/chart-operator/integration/templates"
 )
 
-func DeleteResources(ctx context.Context, helmClient *helmclient.Client, l micrologger.Logger) error {
+func DeleteResources(ctx context.Context, config setup.Config) error {
```
Comment (rossf7) [inline integration/chartconfig/chartconfig.go]: Uses the config struct rather than separate arguments.

File: integration/setup/config.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+	"github.com/giantswarm/micrologger"
+)
+
+type Config struct {
+	Host       *framework.Host
+	HelmClient *helmclient.Client
+	Logger     micrologger.Logger
+	Resource   *resource.Resource
+}
+
```
Comment (rossf7) [inline integration/setup/config.go]: The new setup code.

File: integration/test/chartconfig/basic/basic_test.go [hunk 3]
```diff
@@ -23,7 +23,7 @@ func TestChartLifecycle(t *testing.T) {
 	ctx := context.Background()
 
 	// Setup
-	err := chartconfig.InstallResources(ctx, h, helmClient, l)
+	err := chartconfig.InstallResources(ctx, config)
```
Comment (rossf7) [inline integration/test/chartconfig/basic/basic_test.go]: Uses the config struct in the tests.

File: integration/test/chartconfig/basic/basic_test.go [hunk 4]
```diff
@@ -66,65 +66,65 @@ func TestChartLifecycle(t *testing.T) {
 			VersionBundleVersion: versionBundleVersion,
 		}
 
-		l.Log("level", "debug", "message", fmt.Sprintf("creating %s", cr))
+		config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("creating %#q", cr))
```
Comment (rossf7) [inline integration/test/chartconfig/basic/basic_test.go]: Also fixing the logging.

File: integration/setup/config.go [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+		}
+	}
+
+	var host *framework.Host
+	{
+		c := framework.HostConfig{
+			Logger: logger,
+
+			ClusterID:       "n/a",
+			VaultToken:      "n/a",
```
Comment (xh3b4sd) [inline integration/setup/config.go]: That is so stupid. We should further work on getting rid of the old framework thingy and come up with better designs and functionalities.

Comment (rossf7) [reply integration/setup/config.go]: Yes definitely the current framework abstractions don't fit well here.

**Pull request merged**